### ISSUE TITLE
dev: refactor NN functions

### ISSUE THREAD

**Issue by chachaleo** [gizatechxyz/orion#570]
Making a file containing the common functions and enum used by the nn functions.

**Comment by raphaelDkhn** [comment]
@chachaleo Is this problem related to what you've done [here](https://github.com/gizatechxyz/orion/blob/3f28532a02af4390397cdaf3648ea056329f8b3b/src/operators/nn/common.cairo#L1-L14) in your PR?

**Comment by chachaleo** [comment]
Yes I will gather all enum used by the nn functions in a file and also have a helper file to gather the helper functions (that I will optimize and comment)

**Comment by ScottyDavies** [comment]
@chachaleo I would like to work on this
